### ISSUE TITLE
Keyword with Mode tokenize incorrectly

### ISSUE THREAD

**Issue by bdke** [b3b00/csly#501]
### Info
- Version: 3.4.5

### Token
```cs
[Keyword("import")]
[Push("importMode")]
ImportKeyword,
[UpTo(";")]
[Mode("importMode")]
ImportContent,
[Sugar(";")]
[Mode("importMode")]
[Pop]
EndImport,
[AlphaNumDashId]
Identifier
```

### Input
```
test = [$"test {test + 3}"];
```

### Output
```
Identifier [test] @line 0, column 0 on channel 0
ImportContent [ = [$"test {test + 3}"]] @line 0, column 5 on channel 0
EndImport [;] @line 0, column 28 on channel 0
<<EOS>>
```

**Comment by b3b00** [comment]
@bdke I 'm starting to look at it

**Comment by b3b00** [comment]
@bdke could you please provide what the expected output is ? 
I agree that the output is incorrect. But as long as I understand your lexer, lexer should failed on  `=` after the first `test| keyword.
Are you expecting a failure ?

**Comment by bdke** [comment]
it should be using the `default` mode tokens
but it is using `importMode` tokens after the `Identifier`

**Comment by bdke** [comment]
it should be tokenized without error

**Comment by bdke** [comment]
more accurate tokens would be
```cs
[Keyword("import")]
[Push("importMode")]
ImportKeyword,
[UpTo(";")]
[Mode("importMode")]
ImportContent,
[Sugar(";")]
[Mode("importMode")]
[Pop]
EndImport,

[AlphaNumDashId]
Identifier,
[Int]
Number,
[Sugar("=")]
Assign
```

content: `test = 3`

**Comment by b3b00** [comment]
fixed with version 3.4.6

**Comment by bdke** [comment]
thx
